src/mod/context.rb

```diff
@@ -43,7 +43,7 @@ def primitive(val)
   private :primitive
 
   def status
-    {
+    @status ||= {
```

Comment:
Minor: Since this object gets reused, perhaps worth sprinkling a bunch of `.freeze` to guard against oops?